### PR TITLE
Chat: Fix heading styles and inline code colors

### PR DESCRIPTION
**Pull request by toolmantim**
Improves the rendering of headings and inline code elements in chat.

| Before | After |
| - | - |
| Screenshot 2023-10-27 at 1 52 41 pm | Screenshot 2023-10-27 at 1 52 08 pm |
| Screenshot 2023-10-27 at 1 52 45 pm | Screenshot 2023-10-27 at 1 52 04 pm |
| Screenshot 2023-10-27 at 1 59 05 pm | Screenshot 2023-10-27 at 1 58 53 pm |

## Test plan

- Verified in chat output (using https://gist.github.com/toolmantim/2e426f846f7a668172c7cb15b0051fc9)

### CODE REVIEW

File: lib/ui/src/chat/TranscriptItem.module.css [hunk 1]
```diff
@@ -53,6 +53,30 @@
     overflow-x: auto;
 }
 
+.content h1, .content h2, .content h3, .content h4, .content h5, .content h6 {
```
Comment (toolmantim) [inline lib/ui/src/chat/TranscriptItem.module.css:56]: Screenshot 2023-10-27 at 2 07 58 pm
 
😅

File: vscode/webviews/Chat.module.css [hunk 2]
```diff
@@ -27,46 +27,31 @@ body[data-vscode-theme-kind='vscode-high-contrast'] .human-transcript-item {
...
-
-    margin-bottom: 0;
-}
-
-body[data-vscode-theme-kind='vscode-light'] .transcript-item pre {
-    border-style: solid;
-    border-width: 1px;
-    border-color: var(--vscode-sideBarSectionHeader-border);
-
-    border-bottom: none;
```
Comment (toolmantim) [inline vscode/webviews/Chat.module.css:69]: Removed as they were unnecessarily duped

File: lib/ui/src/chat/TranscriptItem.module.css (hunk 1 above)
Comment (dominiccooney) [reply lib/ui/src/chat/TranscriptItem.module.css:56]: Gotta get the brand out there.

File: lib/shared/src/chat/markdown.ts [hunk 3]
```diff
@@ -37,8 +37,18 @@ const DOMPURIFY_CONFIG = {
...
+        'h2',
+        'h3',
+        'h4',
+        'h5',
+        'h6',
     ],
     ALLOWED_URI_REGEXP,
+    FORBID_ATTR: [
+        // Disallow heading ids, or any ids
+        'id',
```
Comment (dominiccooney) [inline lib/shared/src/chat/markdown.ts:50]: Nice. Maybe we should disallow `name`, I think some ancient DOM collections key on that.

**Review by dominiccooney — APPROVED** [review]
Looks great.

File: lib/shared/src/chat/markdown.ts (hunk 3 above)
Comment (toolmantim) [reply lib/shared/src/chat/markdown.ts:50]: Ah. This was mostly to remove the auto-generated `id` attribute… I didn't begin to dig into if there were other things to add.

**Pull request merged**